### PR TITLE
refs #22301 - upload docker manifest and tag

### PR DESCRIPTION
**Pull request by thomasmckay**

### CODE REVIEW

**Comment by akofink** [comment]
Just some rubocop failures here. It works as expected.

File: test/functional/repository/update_test.rb [hunk 1]
```diff
@@ -103,6 +104,61 @@ module HammerCLIKatello
...
+        repo_id = 3
+        upload_id = "1234"
+        upload_response = {
+          "upload_id" => upload_id,
+          "_href" => "/pulp/api/v2/content/uploads/#{upload_id}"
+        }
+
+        ex = api_expects(:content_uploads, :create, "Create upload for content") do |par|
+          par[:repository_id] == repo_id
+        end
```
[the remark this answers is not in the document]
Comment (thomasmckay) [reply test/functional/repository/update_test.rb]: @akofink - updated, thanks!

File: test/functional/repository/update_test.rb [hunk 2]
```diff
@@ -103,6 +104,60 @@ module HammerCLIKatello
...
+          "upload_id" => upload_id,
+          "_href" => "/pulp/api/v2/content/uploads/#{upload_id}"
+        }
+
+        ex = api_expects(:content_uploads, :create, "Create upload for content")
+             .with_params(:repository_id => repo_id)
+
+        ex.returns(upload_response)
+
+        ex2 = api_expects(:repositories, :import_uploads, 'Take in an upload') do |par|
```
Comment (akofink) [inline test/functional/repository/update_test.rb]: @thomasmckay There are a bunch of instances of the same thing.

**Comment by thomasmckay** [comment]
[test]

File: test/functional/repository/upload_test.rb [hunk 3]
```diff
@@ -59,6 +59,42 @@
     File.delete("test.rpm")
   end
 
+  it "uploads a docker image" do
+    File.new("test.rpm", "w")
+
+    params = ["--id=#{repo_id}", "--path=#{path}"]
+
+    ex = api_expects(:content_uploads, :create, "Create upload for content") do |par|
```
Comment (akofink) [inline test/functional/repository/upload_test.rb]: This file too, @thomasmckay. :)

**Pull request merged**